addons/barcodes/static/tests/float_scannable_field_test.js

```diff
@@ -0,0 +1,161 @@
+/** @odoo-module **/
+
+import { nextTick, triggerEvent, patchWithCleanup, getFixture } from "@web/../tests/helpers/utils";
+import { makeView, setupViewRegistries } from "@web/../tests/views/helpers";
+import { barcodeGenericHandlers } from "@barcodes/barcode_handlers";
+import { barcodeService } from "@barcodes/barcode_service";
+import { FormController } from "@web/views/form/form_controller";
+import { useBus } from "@web/core/utils/hooks";
+import { registry } from "@web/core/registry";
+
+const maxTimeBetweenKeysInMs = barcodeService.maxTimeBetweenKeysInMs;
+
+function simulateBarCode(chars, target, selector) {
+    for (let char of chars) {
+        let keycode;
+        if (char === "Enter") {
+            keycode = $.ui.keyCode.ENTER;
+        } else if (char === "Tab") {
+            keycode = $.ui.keyCode.TAB;
+        } else {
+            keycode = char.charCodeAt(0);
+        }
+        triggerEvent(target, selector, "keydown", {
+            key: char,
+            keyCode: keycode,
+        });
+    }
+}
+
+let target;
+let serverData;
+
+QUnit.module("Fields", (hooks) => {
+    hooks.beforeEach(() => {
+        target = getFixture();
+        barcodeService.maxTimeBetweenKeysInMs = 0;
+        registry.category("services").add("barcode", barcodeService, { force: true });
+        registry
+            .category("services")
+            .add("barcode_autoclick", barcodeGenericHandlers, { force: true });
+
+        serverData = {
+            models: {
+                product: {
+                    fields: {
+                        name: { string: "Product name", type: "char" },
+                        float_field: { string: "Float", type: "float" },
+                        float_field_2: { string: "Float", type: "float" },
+                        barcode: { string: "Barcode", type: "char" },
+                    },
+                    records: [
+                        { id: 1, name: "Large Cabinet", barcode: "1234567890" },
+                        { id: 2, name: "Cabinet with Doors", barcode: "0987654321" },
+                    ],
+                },
+            },
+        };
+        setupViewRegistries();
+    });
+    hooks.afterEach(() => {
+        barcodeService.maxTimeBetweenKeysInMs = maxTimeBetweenKeysInMs;
+    });
+
+    QUnit.module("FloatScannableField");
+
+    QUnit.test("widget field_float_scannable", async function (assert) {
+        serverData.models.product.records[0].float_field = 4;
+
+        patchWithCleanup(FormController.prototype, {
```

Comment:
this is kind of awkward just to get a ref to the barcode bus. i think there should be a better way to get the env.

Maybe something like this:
```js
const formView = await makeView(...);
formView.env.services.barcode.bus.addEventListener('barcode_scanned', ...)
```

Reply:
I realized that I can do `useBus` to I switched to "patching". But I'll go back to what you're suggesting, it's less bizarre.